statsmodels/stats/anova.py

```diff
@@ -362,6 +367,198 @@ def anova_lm(*args, **kwargs):
+                associated with the factor `key`
+            params : OLS solution of y = x * params
+            keys : factors to be excluded
+
+            Returns
+            -------
+            residual sum of squared and degree of freedom
+
+            """
+            ind = _not_slice(term_slices, keys, x.shape[1])
+            params1 = params[ind]
+            ssr = np.subtract(y, x[:, ind].dot(params1))
+            ssr = ssr.T.dot(ssr)
+            df_resid = len(y) - len(params1)
+            return ssr, df_resid
+
+        # Construct OLS endog and exog from string using patsy
+        within = ['C(%s, Sum)' % i for i in self.within]
+        subject = 'C(%s, Sum)' % self.subject
+        factors = within + [subject]
+        x = patsy.dmatrix('*'.join(factors), data=self.data)
+        term_slices = x.design_info.term_name_slices
+        for key in term_slices:
+            ind = np.array([False]*x.shape[1])
+            ind[term_slices[key]] = True
+            term_slices[key] = np.array(ind)
+        term_exclude = [':'.join(factors)]
+        ind = _not_slice(term_slices, term_exclude, x.shape[1])
+        x = x[:, ind]
+
+        # Fit OLS
+        model = OLS(y, x)
+        results = model.fit()
+        if model.rank < x.shape[1]:
+            raise ValueError('Independent variables are collinear.')
+        for i in term_exclude:
+            term_slices.pop(i)
+        for key in term_slices:
+            term_slices[key] = term_slices[key][ind]
+        params = results.params
+        df_resid = results.df_resid
+        ssr = results.ssr
+
+        anova_table = pd.DataFrame(
+            {'F Value':[], 'Num DF':[], 'Den DF':[], 'Pr > F':[]})
+
+        for key in term_slices:
+            if self.subject not in key and key != 'Intercept':
+                #  Independen variables are orthogonal
+                ssr1, df_resid1 = _ssr_reduced_model(
+                    y, x, term_slices, params, [key])
+                df1 = df_resid1 - df_resid
+                msm = (ssr1 - ssr) / df1
+                if key == ':'.join(factors[:-1]) or (key + ':' +
+                                                     subject not in term_slices):
+                    mse = ssr / df_resid
+                    df2 = df_resid
+                else:
+                    ssr1, df_resid1 = _ssr_reduced_model(
+                        y, x, term_slices, params,
+                        [key+ ':' + subject])
+                    df2 = df_resid1 - df_resid
+                    mse = (ssr1 - ssr) / df2
+                F = msm / mse
+                p = stats.f.sf(F, df1, df2)
+                term = key.replace('C(', '').replace(', Sum)', '')
+                anova_table.loc[term, 'F Value'] = F
+                anova_table.loc[term, 'Num DF'] = df1
+                anova_table.loc[term, 'Den DF'] = df2
+                anova_table.loc[term, 'Pr > F'] = p
+        self.anova_table = anova_table.iloc[:, [1, 2, 0, 3]]
```

Comment:
is there a reason to attach `anova_table` to `self`, i.e. the AnovaRM instance.
It would be cleaner to provide it as argument to `ANOVARestults.__init__`.

(Currently irrelevant, but better in case there will be fit options)